Refactor: software engineer job description (#159)
compilerla/compiler.la: _jobs/software-engineer.md

This is a full-time, salaried position (W-2) with benefits. Full-time employment at Compiler is thirty (30) hours per week.

## Responsibilities

- Design and build open source software that streamlines the delivery of public services with a focus on equity and active harm reduction
- Interact with clients to iteratively gather requirements, refine scope, and demo progress
- Document your work, including asynchronous communications, to inform internal and client teams
- Coordinate with product, design, and engineering to deliver delightful products and services that meet current Web Content Accessibility Guidelines (WCAG)
- Pair program, code review, contribute to architecture decisions, help create standard patterns and practices for development and DevOps
- Self-direct and time-manage across multiple client and internal projects with varying levels of complexity, demand, and constraint

## Preferred Qualifications

- Experience working with or in proximity to government
- Experience using modern, high-level programming languages such as Python and JavaScript to build web applications, API services, or similar

> - Experience developing performant and accessible front-end components that meet WCAG standards, using advanced HTML and CSS

Reviewer: ```suggestion
- Experience developing performant and accessible front-end components that meet WCAG standards, using advanced HTML and CSS
- Experience maintaining databases
- Experience working with external APIs
```

Author: These are both great suggestions but I feel like as worded, they aren't saying enough. Can you please add a bit more to each of these new bullets? 

E.g. `Experience integrating production applications with external/3rd party APIs to do X (or for reason Y)`

Reviewer: Updated my suggestion

- Experience collaborating with product managers, UI/UX designers, remote teams, along with client technical teams
- Knowledge of Git / GitHub-based development workflows and open source project management
- A love of public transportation
- A personal and professional commitment to bring anti-racist principles into your work
- Preference given to candidates located in California, and specifically Los Angeles

## Education

- Per company policy, this position has no formal educational requirements

## Start Date

- Approximately Monday, April 1st, 2024

## Compensation

- Expected salary range is $100,000 - $120,000 annually based on qualifications and experience
- Paid time off: fifteen (15) days paid time off (PTO) and ten (10) sick days annually, federal holidays, and annual holiday closure between December 23 and January 1
- 100% Company-paid medical, dental, and vision insurance premiums for employees; 50% Company-paid insurance premiums for employees’ dependents
- 100% Company-paid short- and long-term disability insurance
- Discretionary budget for home office and/or professional development
- Company 401k contribution

> - Company-paid in-person retreats twice a year

Reviewer 2: One paid (required?) conference of your choice

Author: Thanks for this suggestion too @machikoyasuda. We’re trying to turn this into a formal policy but we feel we don’t want to share it with potential candidates as a perk until it’s formalized, as part of our budget process for 2024.

> - Company-provided laptop

Reviewer 2: Parental leave is not on here

Author: Shelby also called this out. It has been added in 9fbeb8f.

## Location and Work Hours

- This position is 100% remote
- Los Angeles-based staff have the opportunity to work from our office at the Los Angeles Cleantech Incubator (LACI) in downtown Los Angeles, CA

> - Flexible working hours; expect an overlap with the Pacific timezone. The current team is spread across Pacific, Central, and Eastern timezones in the US

Reviewer 2: ```suggestion
- Flexible working hours; expect an overlap with the Pacific time zone. The current team is spread across Pacific, Central, and Eastern time zones in the US
```

- Document your work, including asynchronous communications, to inform internal and client teams

> - Coordinate with product, design, and engineering to deliver delightful products and services

Reviewer 2: ```suggestion
- Coordinate with product, design, and engineering to deliver delightful products and services that meet current Web Content Accessibility Guidelines (WCAG)
```

Reviewer 2: I think I'd prefer more objective adjectives like `performant, secure, well-designed` rather than `delightful` - but maybe doesn't matter that much. Definitely want WCAG in there though!

- Pair program, code review, contribute to architecture decisions, help create standard patterns and practices for development and DevOps
- Self-direct and time-manage across multiple client and internal projects with varying levels of complexity, demand, and constraint

## Preferred Qualifications

- Experience working with or in proximity to government
- Experience using modern, high-level programming languages such as Python and JavaScript to build web applications, API services, or similar
- Experience using Docker and containers to manage reproducible environments

> - Experience deploying and managing production applications and services in cloud environments such as Azure and GCP, preferably with Infrastructure as Code systems such as Terraform

Reviewer 2: ```suggestion
- Experience deploying and managing production applications and services in cloud environments such as Azure and GCP, preferably with Infrastructure as Code systems such as Terraform
- Experience and apprecation of writing unit and integration tests for code
```

- Experience collaborating with product managers, UI/UX designers, and remote teams

> - Knowledge of Git / GitHub based development workflows and open source project management

Reviewer 2: ```suggestion
- Knowledge of Git / GitHub-based development workflows and open source project management
```

- A love of public transportation
- A personal and professional commitment to bring anti-racist principles into your work

> - Preference given to candidates located in California, and specifically Los Angeles

Reviewer 2: LA..city? County? Does it matter? @thekaveman

Author: I think we'll leave this one more open. It doesn't matter as much, more so that they are local-ish.
